homeassistant/components/sensor/rest.py

```diff
@@ -0,0 +1,141 @@
+https://home-assistant.io/components/sensor.rest.html
+"""
+import logging
+from requests import get, exceptions
+from datetime import timedelta
+
+from homeassistant.util import Throttle
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "REST Sensor"
+
+# Return cached results if last scan was less then this time ago
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
+
+
+# pylint: disable=unused-variable
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Get the REST sensor. """
+
+    resource = config.get('resource', None)
+
+    try:
+        response = get(resource, timeout=10)
+    except exceptions.MissingSchema:
+        _LOGGER.error("Missing resource or schema in configuration. "
+                      "Add http:// to your URL.")
+        return False
+    except exceptions.ConnectionError:
+        _LOGGER.error("No route to resource/endpoint. "
+                      "Please check the URL in the configuration file.")
+        return False
+
+    rest = RestData(resource)
+
+    dev = []
+    add_devices([RestSensor(rest,
+                            config.get('name', DEFAULT_NAME),
+                            config.get('variable'),
+                            config.get('unit'))])
+    add_devices(dev)
+
+
+class RestSensor(Entity):
+    """ Implements a REST sensor. """
+
+    def __init__(self, rest, name, variable, unit_of_measurement):
+        self.rest = rest
+        self._name = name
+        self._variable = variable
+        self._state = 'n/a'
+        self._unit_of_measurement = unit_of_measurement
+        self.update()
+
+    @property
+    def name(self):
+        """ The name of the sensor. """
+        return self._name
+
+    @property
+    def unit_of_measurement(self):
+        """ Unit the value is expressed in. """
+        return self._unit_of_measurement
+
+    @property
+    def state(self):
+        """ Returns the state of the device. """
+        return self._state
+
+    def update(self):
+        """ Gets the latest data from REST API and updates the state. """
+        self.rest.update()
+        value = self.rest.data
+
+        if 'error' in value:
+            self._state = value['error']
+        else:
+            try:
+                self._state = value[self._variable]
+            except KeyError:
+                _LOGGER.error('Variable "%s" not found in response: "%s".',
+                              self._variable, value)
+                self._state = 'N/A'
+
+
+# pylint: disable=too-few-public-methods
+class RestData(object):
+    """ Class for handling the data retrieval. """
+
+    def __init__(self, resource):
+        self.resource = resource
+        self.data = dict()
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """ Gets the latest data from REST service. """
+        try:
+            response = get(self.resource, timeout=10)
+            if 'error' in self.data:
```

Comment:
> Having an error key or message in the body is less common than a non-200x status. May I suggest:

I'm not checking for an error key in the body of the response, the check is for the return value. The idea behind is that if an error occurs then the error is added to the dict along the values. This way the values are still available and could be used in the attr of the sensor in the frontend as last known values. 

> Also, you might have an option to specify the type of request. Some people might want to do a POST request with some data, but one thing at a time :)

Good point.